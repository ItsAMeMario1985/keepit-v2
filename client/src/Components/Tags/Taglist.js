import PropTypes from 'prop-types'
import styled from 'styled-components/macro'
import LoadingSpinner from '../LoadingSpinner/LoadingSpinner'

import { CloudIcon, UserIcon } from '../Icons'

export default function Taglist({
  onClick,
  tags,
  bgColor,
  showIsCustom,
  showIsloading,
}) {
  Taglist.propTypes = {
    onClick: PropTypes.func,
    tags: PropTypes.array.isRequired,
    bgColor: PropTypes.string,
    showIsCustom: PropTypes.bool.isRequired,
    showIsloading: PropTypes.bool.isRequired,
  }

  function IsCustomIcon(tag) {
    if (showIsCustom) {
      if (tag.isCustom === 'true') {
        return <UserIcon />
      } else {
        return <CloudIcon fill="#fff" />
      }
    } else {
      return ''
    }
  }

  if (tags.length === 0 && showIsloading) {
    return <LoadingSpinner />
  }

  return (
    <StyledUl>
      {tags &&
        tags.map((tag, index) => (
          <StyledLi
            clickable={onClick ? true : false}
            key={tag.value}
            bgcolor={bgColor}
            onClick={
              onClick ? () => onClick(tag.value, !tag.added) : console.log('#')
            }
          >
            <StyledTagText>{tag.value} </StyledTagText>
            <IsCustomIcon isCustom={tag.isCustom}></IsCustomIcon>
          </StyledLi>
        ))}
    </StyledUl>
  )
}

const StyledTagText = styled.span`
  font-weight: 400;
  font-size: 14px;
  letter-spacing: 0.5px;
  margin: 0 8px 0 5px;
`

const StyledUl = styled.ul`
  list-style-type: none;
  color: white;
  display: flex;
  flex-wrap: wrap;
  min-height: 24px;
`

const StyledLi = styled.li`
  background-color: ${(props) => props.bgcolor};
  display: inline-block;
  border-bottom-right-radius: 8px;
  border-top-right-radius: 8px;
  min-width: 50px;
  display: inline-flex;
  align-items: center;
  padding: 5px 10px 5px 5px;
  border-left: 2px solid #e3e3e3;
  cursor: ${(props) => (props.clickable ? 'pointer' : 'auto')};
  font-weight: 400;
  font-size: 14px;
  margin-bottom: 5px;
  margin-right: 5px;
`
