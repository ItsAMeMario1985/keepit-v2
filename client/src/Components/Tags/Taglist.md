```jsx
<TagList
  tags={[{ value: 'TestTag', added: false, isCustom: false }]}
  onClick={() => {}}
></TagList>
```
