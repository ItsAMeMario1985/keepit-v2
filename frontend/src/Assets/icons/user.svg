<svg xmlns="http://www.w3.org/2000/svg" width="13"  viewBox="0 0 19.161 15.329">
  <path id="Icon_awesome-user-edit" data-name="Icon awesome-user-edit" d="M6.706,7.665A3.832,3.832,0,1,0,2.874,3.832,3.832,3.832,0,0,0,6.706,7.665Zm2.683.958h-.5a5.212,5.212,0,0,1-4.365,0h-.5A4.025,4.025,0,0,0,0,12.647v1.245a1.437,1.437,0,0,0,1.437,1.437h8.23a1.438,1.438,0,0,1-.078-.638l.2-1.823.036-.332.237-.237L12.38,9.985A3.983,3.983,0,0,0,9.389,8.623Zm1.356,4.35-.2,1.826a.477.477,0,0,0,.527.527l1.823-.2,4.129-4.129L14.874,8.847l-4.129,4.126Zm8.206-4.922L17.817,6.916a.717.717,0,0,0-1.012,0L15.673,8.048l-.123.123,2.15,2.147,1.251-1.251a.721.721,0,0,0,0-1.015Z" fill="#fff"/>
</svg>
