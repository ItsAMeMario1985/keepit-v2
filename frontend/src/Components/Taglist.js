import PropTypes from 'prop-types'
import styled from 'styled-components/macro'
import loading from '../Assets/loading.gif'

import { CloudIcon, UserIcon } from './Icons'

export default function Taglist({
  onClick,
  tags,
  bgColor,
  showIsCustom,
  showIsloading,
}) {
  Taglist.propTypes = {
    onClick: PropTypes.func.isRequired,
    tags: PropTypes.array.isRequired,
  }

  function IsCustomIcon(tag) {
    if (showIsCustom) {
      if (tag.isCustom) {
        return <UserIcon />
      } else {
        return <CloudIcon />
      }
    } else {
      return ''
    }
  }

  if (tags.length === 0 && showIsloading) {
    return (
      <StyledLoading>
        <img width="20" src={loading}></img>
      </StyledLoading>
    )
  }

  return (
    <StyledTagList bgcolor={bgColor}>
      {tags &&
        tags.map((tag, index) => (
          <li key={tag.value} onClick={() => onClick(tag.value, !tag.added)}>
            <StyledTagText>{tag.value} </StyledTagText>
            <IsCustomIcon isCustom={tag.isCustom}></IsCustomIcon>
          </li>
        ))}
    </StyledTagList>
  )
}

const StyledTagText = styled.span`
  font-weight: 400;
  font-size: 14px;
  letter-spacing: 0.5px;
  margin: 0 8px 0 5px;
`

const StyledTagList = styled.ul`
  list-style-type: none;
  color: white;
  display: flex;
  flex-wrap: wrap;
  gap: 5px;
  min-height: 24px;
  li {
    background-color: ${(props) => props.bgcolor};
    display: inline-block;
    border-bottom-right-radius: 8px;
    border-top-right-radius: 8px;
    min-width: 50px;
    display: inline-flex;
    align-items: center;
    padding: 5px 10px 5px 5px;
    border-left: 2px solid #e3e3e3;
    cursor: pointer;
    font-weight: 400;
    font-size: 14px;
  }
`
const StyledLoading = styled.div`
  text-align: center;
  font-size: 12px;
`
