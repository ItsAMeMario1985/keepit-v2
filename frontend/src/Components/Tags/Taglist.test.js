import { render } from '@testing-library/react'
import Taglist from './Taglist'
import user from '@testing-library/user-event'

const onClickMock = jest.fn()
const props = {
  onClick: onClickMock,
  tags: [
    {
      value: 'Beer',
      added: false,
      isCustom: false,
    },
    {
      value: 'Barware',
      added: false,
      isCustom: false,
    },
  ],
  bgColor: 'var(--color-primary)',
  showIsCustom: false,
  showIsloading: false,
}

describe('Taglist', () => {
  it('renders correctly', () => {
    const { container } = render(<Taglist {...props} />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('onclick works', () => {
    const { getAllByRole } = render(<Taglist {...props} />)
    const listItem = getAllByRole('listitem')
    user.click(listItem[0])
    expect(onClickMock).toHaveBeenCalledTimes(1)
  })
})
