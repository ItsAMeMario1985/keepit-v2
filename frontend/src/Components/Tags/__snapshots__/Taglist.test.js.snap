// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Taglist renders correctly 1`] = `
<ul
  class="Taglist__StyledUl-e2yq2l-1 hXgrG"
>
  <li
    class="Taglist__StyledLi-e2yq2l-2 bMScUO"
  >
    <span
      class="Taglist__StyledTagText-e2yq2l-0 iyvejS"
    >
      Beer
       
    </span>
    
  </li>
  <li
    class="Taglist__StyledLi-e2yq2l-2 bMScUO"
  >
    <span
      class="Taglist__StyledTagText-e2yq2l-0 iyvejS"
    >
      Barware
       
    </span>
    
  </li>
</ul>
`;
